apps/game/src/sandbox/time_warp.rs

```diff
@@ -421,7 +429,14 @@ fn area_under_curve(raw: Vec<(Time, usize)>, width: f64, height: f64) -> Polygon
     }
     downsampled.push(Pt2D::new(width, height));
     downsampled.push(downsampled[0]);
-    Ring::must_new(downsampled).into_polygon()
+
+    Ring::deduping_new(downsampled)
+        .map_or_else(|e| {
+            Err(e)
+        },
+        |ring| {
+            Ok(ring.into_polygon())
+        })
```

Comment:
```suggestion
Ring::deduping_new(downsampled).map(|ring| ring.into_polygon())
```

This is a simpler way to write the same thing. We don't need to transform the error type in this case, because `deduping_new` already returns anyhow's Result type. So we just need to transform the successful case

Reply:
ahh darn! initially had this but had compiler errors since i was using anyhow::error